Extension/src/Debugger/debugAdapterDescriptorFactory.ts

```diff
@@ -33,21 +33,20 @@ export class CppdbgDebugAdapterDescriptorFactory extends AbstractDebugAdapterDes
         super(context);
     }
 
-    createDebugAdapterDescriptor(session: vscode.DebugSession, executable?: vscode.DebugAdapterExecutable): vscode.ProviderResult<vscode.DebugAdapterDescriptor> {
-        return util.isExtensionReady().then(ready => {
-            if (ready) {
-                let command: string = path.join(this.context.extensionPath, './debugAdapters/OpenDebugAD7');
-
-                // Windows has the exe in debugAdapters/bin.
-                if (os.platform() === 'win32') {
-                    command = path.join(this.context.extensionPath, "./debugAdapters/bin/OpenDebugAD7.exe");
-                }
+    async createDebugAdapterDescriptor(session: vscode.DebugSession, executable?: vscode.DebugAdapterExecutable): Promise<vscode.ProviderResult<vscode.DebugAdapterDescriptor>> {
```

Comment:
I seem to remember talking to you about this one. You can't change the return type for this because it's implementing an abstract method.  I'm surprised that this is compiling in CI.  We should just revert this one and let it keep using `.then`.

Reply:
Bob, I think you may be mistaken.  You can provide an overload with a compatible type with the abstract function's return type.  ProviderResult<T> is:
```
export type ProviderResult<T> = T | undefined | null | Thenable<T | undefined | null>;
```

So, this could return a `Promise<vscode.DebugAdapterDescriptor>`, as it's compatible with `Thenable<vscode.DebugAdapterDescriptor>`, and can also be async.  The same applies for various other provider API overloads.  For example, look at `provideCodeActions` in `CodeActionProvider`.  It's async and returns a `Promise<(vscode.Command | vscode.CodeAction)[]>`, yet the API function it's overloading returns a `ProviderResult<(Command | CodeAction)[]>`.

Reply:
You're right.  I think `Promise<vscode.DebugAdapterDescriptor>` is the way to go.